app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -147,7 +148,7 @@ open class HomeActivity : AppCompatActivity() {
         // Toggle off the open_link_in_private_tab pref if we are no longer set as the default browser
         // We do this on a separate thread to alleviate performance issues
         val weakReferenceContext = WeakReference(this)
-        lifecycleScope.launch {
+        lifecycleScope.launch(Dispatchers.IO) {
             val context = weakReferenceContext.get() ?: return@launch
             if (!Browsers.all(context).isDefaultBrowser) {
                 context.settings().preferences
```

Comment:
Does SharedPrefs actually run on the IO thread? I have no idea, but I've seen SharedPrefs-related strict mode problems.

Reply:
afaik, SharedPrefs executes on the thread that you call it on. afaik, it's thread safe – so you can use it from multiple threads – but they use different wording. From SharedPrefs reference docs:

> Note: This class provides strong consistency guarantees. It is using expensive operations which might slow down an app. Frequently changing properties or properties where loss can be tolerated should use other mechanisms. For more details read the comments on SharedPreferences.Editor.commit() and SharedPreferences.Editor.apply().

My concern is that the Settings class was not designated as thread safe by the developers who wrote it so, in the future, it may not enforce the thread safety SharedPreferences has.

Reply:
I just wanted to add to this discussion:

@MarcLeclair and I were hoping that we could move execution of GleanMetrics.setStartupMetrics() off of the main thread and that would also involve using the Settings class off the main thread. I think that means we will need to come to a conclusion about whether or not we can use Settings off MT. 

I think that there are two related issues:
1. Whether Settings can be used off MT (SharedPrefs itself is usable off MT, iirc)
2. Whether Setting is thread safe (SharedPrefs itself is thread safe, iirc)

Reply:
If we identify that Settings is thread safe, imo it's within our authority to add a comment explaining that it's thread safe and therefore making others enforce those guarantees. :)